wallet/wallet-controller/src/lib.rs

```diff
@@ -618,6 +647,31 @@ impl<T: NodeInterface + Clone + Send + Sync + 'static, W: WalletEvents> Controll
             }
 
             tokio::time::sleep(NORMAL_DELAY).await;
+
+            self.rebroadcast_txs(&mut rebroadcast_txs_timer);
+        }
+    }
+
+    /// Rebroadcast not confirmed transactions
+    fn rebroadcast_txs(&mut self, rebroadcast_txs_timer: &mut Instant) {
+        if Instant::now() >= *rebroadcast_txs_timer {
+            let res = self.wallet.get_transactions_to_be_broadcasted().map(|txs| async {
+                for tx in txs {
+                    let tx_id = tx.transaction().get_id();
+                    let res = self.rpc_client.submit_transaction(tx).await;
+                    if let Err(e) = res {
+                        log::error!("Rebroadcasting for tx {tx_id} failed: {e}");
+                    }
+                }
+            });
+
+            if let Err(e) = res {
+                log::error!("Fetching transactions for rebroadcasting failed: {e}");
+            }
```

Comment:
We have the `log_err_pfx` helper for this sort of thing in the `utils` crate

Reply:
changed